src/wkt/src/duration.rs

```diff
@@ -272,6 +272,28 @@ impl std::convert::From<Duration> for time::Duration {
     }
 }
 
+/// Converts from [chrono::Duration] to [Duration].
+#[cfg(feature = "chrono")]
+impl std::convert::TryFrom<chrono::Duration> for Duration {
+    type Error = DurationError;
+
+    fn try_from(value: chrono::Duration) -> Result<Self, Self::Error> {
+        Self::new(value.num_seconds(), value.subsec_nanos())
+    }
+}
+
+/// Converts from [Duration] to [chrono::Duration].
+#[cfg(feature = "chrono")]
+impl std::convert::TryFrom<Duration> for chrono::Duration {
+    type Error = DurationError;
+    fn try_from(value: Duration) -> Result<Self, Self::Error> {
+        if value.nanos < 0 {
+            return Err(Error::OutOfRange());
+        }
+        Ok(Self::new(value.seconds(), value.nanos() as u32).unwrap())
+    }
```

Comment:
I think you can safely handle the `value.nanos < 0` case using:

```suggestion
        Ok(Self::seconds(value.seconds()) + Self::nanoseconds(value.nanos()))
    }
```

Which suggests you do not need a `TryFrom` at all, and a simple `From` will do.

Reply:
That's a good idea!